flink-runtime/src/main/java/org/apache/flink/runtime/taskexecutor/TaskManagerRunner.java

```diff
@@ -366,11 +367,15 @@ public static TaskExecutor startTaskManager(
 			resourceID,
 			taskManagerServicesConfiguration.getSystemResourceMetricsProbingInterval());
 
+		final ExecutorService ioExecutor = Executors.newCachedThreadPool(
+			taskManagerServicesConfiguration.getNumIoThreads(),
+			new ExecutorThreadFactory("flink-taskexecutor-io"));
```

Comment:
This could be moved into the `fromConfiguration` method. Given that the `TaskManagerServices` is responsible for managing the created `ExecutorService`, I think it is fine to move the creation into the `fromConfiguration` method.

Reply:
initially that was the plan, but the `TaskExecutorLocalStateStoresManagerTest` passes in a `Executors.directExecutor()` into `TaskManagerServices.fromConfiguration`.

Reply:
although this doesn't seem related to the correctness of the test.

Reply:
Jup. It is also a bit odd that `TaskExecutorLocalStateStoresManagerTest.testCreationFromConfig` needs to create a `TaskManagerServices` instance in order to test the `TaskExecutorLocalStateStoresManager`. I think one could test the same if one factored the creation of the `TaskExecutorLocalStateStoresManager` out into a separate method which is called by `TaskManagerServices.fromConfiguration()`.

Reply:
But we can also keep it as is. The important aspect is that everyone knows that `fromConfiguration` takes ownership over the passed `ioExecutor`.